internal/error.go

```diff
@@ -119,6 +119,9 @@ Workflow consumers will get an instance of *WorkflowExecutionError. This error w
 */
 
 type (
+	// Category of the error. Maps to logging/metrics behaviours.
+	ApplicationErrorCategory string
```

Comment:
Is `ApplicationErrorCategory` intended to be an enum?

Reply:
Yes